### PR TITLE
Perfecting the draft Implementation

### PR DESCRIPTION
**Pull request by imfaith-hq**
New video showing that the draft UI 
https://user-images.githubusercontent.com/59166014/136818881-c96f0b92-2dc0-4040-862f-ee9fb0eab4a1.mp4

https://user-images.githubusercontent.com/59166014/136670804-f03f90af-bd4c-4b96-87cb-2fb4b4ea6e43.mp4

### CODE REVIEW

File: lib/ui/nav_pages/home_page/widgets/home_list_items.dart [hunk 1]
```diff
@@ -15,17 +15,37 @@ import '../home_page_viewmodel.dart';
...
 final navigationService = locator<NavigationService>();
 
-class ThreadTextAndIcon extends StatelessWidget {
+class ThreadTextAndIcon extends ViewModelWidget<HomePageViewModel> {
   const ThreadTextAndIcon({Key? key}) : super(key: key);
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, viewModel) {
     return _TextAndIcon(
```
Comment (jeremiahseun) [inline lib/ui/nav_pages/home_page/widgets/home_list_items.dart:23]: error • The named parameter 'actionsAlignment' isn't defined • lib/ui/view/draft/app/show_alert_dialog_widget.dart:50:5 • undefined_named_parameter

File: lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart [hunk 2]
```diff
@@ -14,19 +14,21 @@ class ExpandableTextFieldScreen extends HookWidget {
...
+
   final keyboardVisibilityController = KeyboardVisibilityController();
   Stream<bool> get stream =>
       keyboardVisibilityController.onChange.asBroadcastStream();
 
   @override
   Widget build(BuildContext context) {
     double maxSize = MediaQuery.of(context).size.height - kToolbarHeight - 30;
-    TextEditingController textController = useTextEditingController();
     return ViewModelBuilder<ExpandableTextFieldScreenViewModel>.reactive(
```
Comment (jeremiahseun) [inline lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart:32]: Viewmodel should be with the build context and not inside

File: lib/ui/view/draft/draft_view.dart [hunk 3]
```diff
@@ -1,25 +1,124 @@
...
   const DraftView({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return ViewModelBuilder<DraftViewModel>.reactive(
+      onModelReady: (model) {
+        model.drafts;
+      },
       builder: (context, model, child) => Scaffold(
         appBar: AppBar(
```
Comment (jeremiahseun) [inline lib/ui/view/draft/draft_view.dart:20]: Use ZuriAppBar

**Review by jeremiahseun — CHANGES_REQUESTED** [review]
Nice work. Please make changes

File: lib/ui/view/draft/app/data_holder_class.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+class Objects {
```
Comment (pauleke65) [inline lib/ui/view/draft/app/data_holder_class.dart:2]: Why is this class named ```Objects```?
also move this file to the models directory where all the model are and name it properly

File: lib/ui/view/threads/thread_detail/thread_detail_viewmodel.dart [hunk 5]
```diff
@@ -78,11 +90,71 @@ class ThreadDetailViewModel extends BaseViewModel {
...
-  void exitPage() {
-    _navigationService.back();
-  }
-
   String time(String timeStamp) {
     return DateFormat.Hm().format(DateTime.parse(timeStamp));
   }
+
+  //draft implementations
+  //TODO - routing to ThreadDetails needs more fixing, the required parameter userPost of class UserPost proves to difficult access from the draft view.
```
Comment (pauleke65) [inline lib/ui/view/threads/thread_detail/thread_detail_viewmodel.dart:98]: Noted

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Great Work👍🏽, just
[] Make all the requested changes
[] Fix Flutter Analyze issues
[] Fix Merge conflicts
[] Redo the UI of the drafts screen to match the design on Figma [Here] (https://www.figma.com/file/dYssjKRVESlcKQYOmieg3a/HNG-5?node-id=6%3A23358)

File: lib/ui/view/draft/draft_view.dart (hunk 3 above)
Comment (imfaith-hq) [reply lib/ui/view/draft/draft_view.dart:20]: done

File: lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart (hunk 2 above)
Comment (imfaith-hq) [reply lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart:32]: I didn't get you really? but then, The file was never within my scope, all I did there was to remove the textEditing controller and pass in another controller from its parent widget for my implementation

File: lib/ui/nav_pages/home_page/widgets/home_list_items.dart (hunk 1 above)
Comment (imfaith-hq) [reply lib/ui/nav_pages/home_page/widgets/home_list_items.dart:23]: I'ts actually defined from its source code when I checked and its throwing no error, I really don't understand what's going on. the class AlertDialog, actually gives a defined parameter actionsAlignment

File: lib/ui/view/draft/app/data_holder_class.dart (hunk 4 above)
Comment (imfaith-hq) [reply lib/ui/view/draft/app/data_holder_class.dart:2]: done

File: lib/ui/nav_pages/home_page/widgets/home_list_items.dart (hunk 1 above)
Comment (imfaith-hq) [reply lib/ui/nav_pages/home_page/widgets/home_list_items.dart:23]: Resolved

File: lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart (hunk 2 above)
Comment (imfaith-hq) [reply lib/ui/shared/smart_widgets/expandable_textfield/expandable_textfield_screen.dart:32]: Resolved

File: lib/ui/view/draft/draft_viewmodel.dart [hunk 6]
```diff
@@ -1,11 +1,189 @@
...
+import 'package:hng/app/app.logger.dart';
 
 class DraftViewModel extends BaseViewModel {
-  final _title = Draft;
-  String get title => _title;
+  final _navigationService = locator<NavigationService>();
+  final connectivityService = locator<ConnectivityService>();
+  final _storageService = locator<SharedPreferenceLocalStorage>();
+  final snackBar = locator<SnackbarService>();
+  final log = getLogger("Draft Page View Model");
```
Comment (jeremiahseun) [inline lib/ui/view/draft/draft_viewmodel.dart:20]: There should be no space
It should be DraftPageViewModel not the spaced one

**Review by jeremiahseun — CHANGES_REQUESTED** [review]
Please make changes

File: lib/ui/view/draft/draft_viewmodel.dart (hunk 6 above)
Comment (imfaith-hq) [reply lib/ui/view/draft/draft_viewmodel.dart:20]: lol, done boss

Comment (imfaith-hq) [reply lib/ui/view/draft/draft_viewmodel.dart]: done

**Review by pauleke65 — APPROVED** [review]
LG

**Pull request merged**